### PR TITLE
import six

### PR DESCRIPTION
**Pull request by cclauss**
## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is PEP 8 compliant
- [x] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [x] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [x] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

These four Python files use the __six__ module but do not import it so this PR adds __import six__.  Also:
* Removed unused imports
* Added __import ctypes__ where required
* Ran isort on imports
* Changed __file()__ to __open()__ because __file()__ was removed in Python 3.

### CODE REVIEW

**Comment by jklymak** [comment]
What is this trying to fix?  All the tests run w/o these changes, so are you saying there is a code use that is untested that requires `import six` and the other changes in these modules?   I haven't looked into this at all, but I'm curious why this is being requested...

**Comment by cclauss** [comment]
We are trying to achieve compatibility with Python 3.

Try __python3 -m flake8 . --count --select=F821__ to see the problems that we are attempting to solve.

**Comment by WeatherGod** [comment]
matplotlib has been working in python3 for a few years now. Are there
specific bugs you have found?

**Comment by efiring** [comment]
He is finding bugs in code that is not normally exercised.  Some of that code should be removed: specifically, the rc_traits_sgskip.py example is left over from 11 years ago when we were exploring the possibility of using Enthought traits.

**Comment by cclauss** [comment]
See https://github.com/timothycrosley/isort#how-does-isort-work

File: lib/matplotlib/backends/backend_qt5cairo.py [hunk 1]
```diff
@@ -1,5 +1,9 @@
-from .backend_cairo import cairo, FigureCanvasCairo, RendererCairo
-from .backend_qt5 import QtCore, QtGui, _BackendQT5, FigureCanvasQT
+from ctypes.c_long import from_address as c_long_from_address
```
Comment (anntzer) [inline lib/matplotlib/backends/backend_qt5cairo.py]: that import won't actually work.

File: lib/matplotlib/backends/backend_qt5cairo.py [hunk 2]
```diff
@@ -23,7 +27,7 @@ def paintEvent(self, event):
         # Adjust the buf reference count to work around a memory leak bug in
         # QImage under PySide on Python 3.
         if QT_API == 'PySide' and six.PY3:
-            ctypes.c_long.from_address(id(buf)).value = 1
+            c_long_from_address(id(buf)).value = 1
```
Comment (tacaswell) [inline lib/matplotlib/backends/backend_qt5cairo.py]: What I meant by 'local imports' was

```python
If QT_API == 'PySide' and six.PY3:
    import ctypes
    ctypes.c_long.from_address(id(buf)).value = 1
```

Comment (tacaswell) [reply lib/matplotlib/backends/backend_qt5cairo.py]: Sorry for not being clean @cclauss !

**Comment by cclauss** [comment]
I have reverted to the original intent: add __import six__ to four files.
* Removed __import ctypes__
* Removed __file()__ --> __open()__
* Removed isort sorting of imports

**Comment by cclauss** [comment]
@tacaswell @anntzer Are you OK with this reworked version?

**Comment by QuLogic** [comment]
Should just remove `_is_list_like`; it's unused and a private.

**Pull request merged**